core/common/src/main/java/alluxio/util/ConfigurationUtils.java

```diff
@@ -504,22 +504,23 @@ public static GetConfigurationPResponse loadConfiguration(InetSocketAddress addr
   }
 
   /**
-   * Loads client scope properties from the property list returned by grpc.
+   * Filters and loads properties contains certain scope from the property list returned by grpc.
    *
    * @param properties the property list returned by grpc
+   * @param scope the scope to filter the received property list
    * @param logMessage a function with key and value as parameter and returns debug log message
    * @return the loaded properties
    */
-  private static Properties loadClientProperties(List<ConfigProperty> properties,
-      BiFunction<PropertyKey, String, String> logMessage) {
+  private static Properties filterAndLoadProperties(List<ConfigProperty> properties,
+      Scope scope, BiFunction<PropertyKey, String, String> logMessage) {
     Properties props = new Properties();
     for (ConfigProperty property : properties) {
       String name = property.getName();
       // TODO(binfan): support propagating unsetting properties from master
       if (PropertyKey.isValid(name) && property.hasValue()) {
         PropertyKey key = PropertyKey.fromString(name);
-        if (!GrpcUtils.contains(key.getScope(), Scope.CLIENT)) {
-          // Only propagate client properties.
+        if (!GrpcUtils.contains(key.getScope(), scope)) {
+          // Only propagate properties contains the target scope
```

Comment:
If a property has scope `ALL` we want that included along with properties from the `CLIENT` scope, correct?

Reply:
I add a description to only allow Scope=WORKER/CLIENT. 
Not sure about the SCOPE=ALL result

Reply:
My understanding is that for properties marked with `Scope.CLIENT` and `Scope.ALL` should be propagated to the client when loading the cluster default values  - if only one is defined in the function, then we could be missing properties?

Reply:
I guess I also don't fully understand what `GrpcUtils.contains` actually does

Reply:
I tested this function, if given scope == CLIENT,
Scope == CLIENT & ALL will be propagated to client
if given scope == WORKER, Scope == WORKER/SERVER/ALL will be propagated to worker.

Reply:
got it. thanks!